docs/source/config.rst

```diff
@@ -0,0 +1,97 @@
+.. Copyright 2022 CMakePP
+..
+.. Licensed under the Apache License, Version 2.0 (the "License");
+.. you may not use this file except in compliance with the License.
+.. You may obtain a copy of the License at
+..
+.. http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing, software
+.. distributed under the License is distributed on an "AS IS" BASIS,
+.. WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+
+#################
+Configuring CMinx
+#################
+
+By default the reST files generated by CMinx are to the authors' liking. We
+realize that preferences will vary from user to user and have thus attempted
+to expose as many customization points as possible. :ref:`default_config` lists
+all currently recognized CMinx options and their default values. Users
+can provide configuration files which set values for one or more of these
+options in order to override the default values.
+
+************************
+About Configuration File
+************************
+
+Configuration files are `YAML <https://yaml.org/>`__ files. Under the hood,
+CMinx relies on the
+`Confuse <https://confit.readthedocs.io/en/latest/#>`__ library for locating,
+loading, and managing the configuration file. Users can directly provide
+CMinx with the path to their configuration file or place their configuration
+file in one of the paths searched by Confuse (*e.g.*, ``/etc/cminx/`` or
+``~/.config/cminx/``). See
+`here <https://confuse.readthedocs.io/en/latest/usage.html#search-paths>`__ for
+more details regarding how Confuse locates configuration files.
+
+For organizational and namespacing purposes we have partitioned the
+options recognized by CMinx into four categories:
+
+1. ``input``.   Affects which files and file parts CMinx will parse.
+2. ``output``.  Affects how CMinx outputs results.
+3. ``logging``. Options pertaining to CMinx's log messages.
```

Comment:
Should probably mention that this takes any options recognized by Python's logging configuration system, probably add a link to [Logging config Dict Schema](https://docs.python.org/3/library/logging.config.html#logging-config-dictschema) and explain that the Yaml section is considered analogous to a dictionary

Reply:
Would you be able to write it? I'm not actually familiar with logging module configurations.